Add Roadmap to Repo (#2388)
AzureAD/microsoft-authentication-library-for-js: roadmap.md

# Roadmap

> **Deliverable**|**Status**|**Expected Delivery**|**Description**

Reviewer: Can we sort the entries by Expected Delivery?

Reviewer 2: +1

Author: will do

:-----:|:-----:|:-----:|:-----:
Broker In Browser |In Progress|TBD|Application model for application nested in iframes, to support silent token acquisition where 3p cookies are blocked. 

> Msal Node Public Preview|In Progress|October 2020|Our Msal Node library is currently in private preview.  Public preview should include full Public Client and Confidential Client support.  When released, we will be commiting to supporting this library at a production level, resolving bugs in a timely mananer. 

Reviewer 3: ```suggestion
Msal Node Public Preview|In Progress|October 2020|Our Msal Node library is currently in private preview.  Public preview should include full Public Client and Confidential Client support.  When released, we will be committing to supporting this library at a production level, resolving bugs in a timely manner. 
```

> Msal Node Public Preview|In Progress|October 2020|Our Msal Node library is currently in private preview.  Public preview should include full Public Client and Confidential Client support.  When released, we will be committing to supporting this library at a production level, resolving bugs in a timely manner. 

Reviewer 4: Link to node read me in github?

SN + I Support In Msal Node |Not Started|December 2020|Explanation of the flow from our python library: https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/60

> Refresh Token Proof of Possesion [Browser]|Not started|December 2020|Proof of Possesion is a feature that enables cryptographically locking a token to a browser.  This feature has already been shipped for Access Tokens.

Reviewer 3: ```suggestion
Refresh Token Proof of Possession [Browser]|Not started|December 2020|Proof of Possession is a feature that enables cryptographically locking a token to a browser.  This feature has already been shipped for Access Tokens.
```

> Tenant Profile Support |Not Started|December 2020|Support for tenant profiles in our cache and account management

Reviewer 3: ```suggestion
Tenant Profile Support |Not Started|December 2020|Support for tenant profiles in our cache and account management.
```

> Msal Angular 2.0 [Support for Auth Code Flow]|In Progress|December 2020|Upgrading our msal angular library to use the msal-browser lib we recently GA'd to support the Auth Code Flow in the browser.  Helps alleviate some issues caused by blocked 3p cookies. 

Reviewer 3: ```suggestion
Msal Angular 2.0 [Support for Auth Code Flow]|In Progress|December 2020|Upgrading our Msal Angular library to use the Msal Browser lib we recently GA'd to support the Auth Code Flow in the browser.  Helps alleviate some issues caused by blocked 3p cookies. 
```

> Msal Angular 2.0 [Support for Auth Code Flow]|In Progress|December 2020|Upgrading our Msal Angular library to use the Msal Browser lib we recently GA'd to support the Auth Code Flow in the browser.  Helps alleviate some issues caused by blocked 3p cookies. 

Reviewer 5: Can we split this into public/private preview?

Author: I am curious what difference we see. Honestly my hope is that we can ship the whole thing at the end of the quarter and potential have some betas ready to test before hand.

Reviewer 4: I still think we should split it to set customer expectations!

> Msal React Private Preview|In Progress|November 2020|Release our Msal React wrapper preview, using auth code flow in browser. 

Reviewer 3: ```suggestion
Msal React Private Preview|In Progress|November 2020|Release our Msal React wrapper preview, using Auth Code Flow in browser. 
```

> Msal React Public Preview|Not Started|TBD|Once we are ready to support out msal react library in production we will move to public preview. 

Reviewer 3: Out vs. our?
```suggestion
Msal React Public Preview|Not Started|TBD|Once we are ready to support our msal react library in production we will move to public preview. 
```

> Msal Node GA|Not Started|January 2021|After public preview, we will work to resolve issues and stabilize the library to be able to announce General Availibilty 

Reviewer 3: ```suggestion
Msal Node GA|Not Started|January 2021|After public preview, we will work to resolve issues and stabilize the library to be able to announce General Availability. 
```
